### ISSUE TITLE
hyperparameter pruning 

### ISSUE THREAD

**Issue by Aslizy** [UKPLab/sentence-transformers#351]
Hi, 

Thanks for the great work! I have been using the multilingual sentence-bert pretrained model and fine-tuning on QAs from conversations with the multipleNegativeRankingLoss for an answer suggestion system. It is working pretty well, however, I am curious whether we can do some hyperparameter pruning easily, for example, relying on Optuna library to find the best combination of lr, weight_decay, optimizer type, etc. Your experience on hyperparameter pruning is appreciated.

**Comment by nreimers** [comment]
Hi @Aslizy 
thanks 👍 

The most important parameter for multipleNegativeRankingLoss  is in my experience the batch size. Select it as large as possible as the significantly make the task harder and the model learns better.

Other hyperparameters like LR, weight decay, optimizer type had in my experiments rather low impacts (unless man selected them so badly that the model failed to learn something).

Best
Nils Reimers

**Comment by PhilipMay** [comment]
See my (not yet merged) PR #327 where I implemented a callback to do exactly what you talk about. Using Optuna and early stopping.
